docs/DataRequirements.md

```diff
@@ -0,0 +1,316 @@
+**their priority for the initial phase of implementation** and latest
+availability status, as far as we could investigate in the Sovereign SDK
+codebase. They are ordered from highest to lowest impact roughly, i.e., the last
+endpoint in the list is the least important and least frequently required.
+
+## Table of Contents
+
+- [Hermes IBC Relayer Data Availability Requirements](#hermes-ibc-relayer-data-availability-requirements)
+  - [Changelog](#changelog)
+  - [Context](#context)
+  - [Table of Contents](#table-of-contents)
+  - [Sequencer RPC](#sequencer-rpc)
+    - [`/sequencer_submitTxs`](#sequencer_submittxs)
+    - [`sequencer_txStatus`](#sequencer_txstatus)
+    - [`/sequencer_health`](#sequencer_health)
+  - [Rollup RPC](#rollup-rpc)
+    - [`/ledger_searchTx`](#ledger_searchtx)
+    - [`/prover_aggregatedProof*`](#prover_aggregatedproof)
+    - [`/ledger_rollupParams`](#ledger_rollupparams)
+    - [`/ledger_rollupStatus`](#ledger_rollupstatus)
+    - [`/rollup_health`](#rollup_health)
+    - [IBC modules RPC](#ibc-modules-rpc)
+      - [Channel endpoints](#channel-endpoints)
+      - [Client endpoints](#client-endpoints)
+      - [Connection endpoints](#connection-endpoints)
+  - [Rollup WebSocket](#rollup-websocket)
+    - [`/ledger_subscribeAggregatedProof`](#ledger_subscribeaggregatedproof)
+    - [`/ledger_subscribeSlots`](#ledger_subscribeslots)
+
+## Sequencer RPC
+
+### `/sequencer_submitTxs`
+
+- Objective:
+  - For submitting batch of transactions into the mempool.
+  - To simulate transaction sending and conduct basic pre-send
+    checks on factors like transaction size and gas fees, etc.
+
+- Priority: High
+
+- Status:
+  - Nothing available to handle (accept and submit) a batch of transactions in
+    one go.
+  - There is a [`sequencer_acceptTx`](https://github.com/Sovereign-Labs/sovereign-sdk/blob/190863c29835af9090e38d79284b24406c33758c/full-node/sov-sequencer/src/lib.rs#L56-L64) method, which stores a transaction into the mempool.
+  - Also
+  [`sequencer_publishBatch`](https://github.com/Sovereign-Labs/sovereign-sdk/blob/cca1729445741aadbec2490c14ca2090afdc878b/full-node/sov-sequencer/src/lib.rs#L74-L90)
+  method for submitting batch of transactions to the DA layer which works in an
+  async fashion.
+
+### `sequencer_txStatus`
+
+- Objective:
+  - Used to check the submission and commitment status of pending transactions
+    on the DA layer, so can decide on transaction re-submission if necessary.
+
+- Priority: Low
+
+- Status: Nothing available yet.
+
+### `/sequencer_health`
+
+- Objective:
+  - Needed for basic check to assess the health of sequencer node.
+  - Only used once, at relayer startup during health check.
+  
+- Priority: Low
+
+- Status: Available as the
+  [`/health`](https://github.com/Sovereign-Labs/sovereign-sdk/blob/1adbfc963bb930edfa0efe6030262dfb70acf199/module-system/sov-modules-macros/src/rpc/rpc_gen.rs#L339-L343)
+  method to check the health of the RPC server.
+
+## Rollup RPC
+
```

Comment:
We will probably need some `/query_account` RPC to get the account info.
In Cosmos chains this returns:
```
pub struct BaseAccount {
    #[prost(string, tag = "1")]
    pub address: ::prost::alloc::string::String,
    #[prost(message, optional, tag = "2")]
    pub pub_key: ::core::option::Option<super::super::super::google::protobuf::Any>,
    #[prost(uint64, tag = "3")]
    pub account_number: u64,
    #[prost(uint64, tag = "4")]
    pub sequence: u64,
}
```

Reply:
Not an RPC but we also need other information like the address type, the public and private key generation, etc.
See `*_key_pair.rs` file in `https://github.com/informalsystems/hermes/tree/master/crates/relayer/src/keyring` used for Cosmos, Solana, Ethermint

Reply:
Added: [4a5146e](https://github.com/informalsystems/sovereign-ibc/pull/57/commits/4a5146e03ef26ee4b08a5f2ddc1b23bc9e8a7337)

As for the `/query_account`, there is an [`accounts_getAccount`](https://github.com/informalsystems/sovereign-sdk/blob/d42e289f26b9824b5ed54dbfbda94007dee305b2/module-system/module-implementations/sov-accounts/src/query.rs#L26-L45) RPC endpoint which appears to perform the same job. However, it accepts a public key of the `C::PublicKey` and returns the `C::Address` type. Meaning that the concrete account type depends on the `C` (rollup context) that would be defined by the SDK implementors. For now, we can rely on their default ready-to-go definitions for such types. Therefore, We've got the [Address](https://github.com/Sovereign-Labs/sovereign-sdk/blob/e6807ed67259d2b82fd6143db42f70357e490a8b/module-system/sov-modules-core/src/common/address.rs#L38-L44), [`DefaultPublicKey`](https://github.com/Sovereign-Labs/sovereign-sdk/blob/df169be87524d8157fd93d3948ac30b21b288719/module-system/sov-modules-api/src/default_signature.rs#L169-L177), and [`DefaultPrivateKey`](https://github.com/Sovereign-Labs/sovereign-sdk/blob/df169be87524d8157fd93d3948ac30b21b288719/module-system/sov-modules-api/src/default_signature.rs#L35-L40) of the `ed25519` type.